### PR TITLE
[CORE-3028] fixed usage of computed values in prepared insert/update changes

### PR DESCRIPTION
**Pull request by Plunts**
## Environment
**Liquibase Version**: 3.5.3 - 4.3.0
**Liquibase Integration & Version**: all
**Liquibase Extension(s) & Version**: n/a
**Database Vendor & Version**: all, tested with mariadb
**Operating System Type & Version**: all

## Pull Request Type
* \[x] Bug fix (non-breaking change which fixes an issue.)
* \[ ] Enhancement/New feature (non-breaking change which adds functionality)
* \[ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Insert and Update-Changes can not combine LOBs with computed column values. This happens because liquibase switches to PreparedStatements to transfer the LOB data, but then also tries to bind the database function. See JIRA issue CORE-3028 for a detailed example.

## Steps To Reproduce
See [CORE-3028](https://liquibase.jira.com/browse/CORE-3028) for a changelog-xml or the included unit test.

## Actual Behavior
Liquibase inserts null to all columns with computed values.

## Expected/Desired Behavior
Liquibase executes the specified database function.

## Fast Track PR Acceptance Checklist:
* \[x] Build is successful and all new and existing tests pass
* \[x] Added [Unit Test(s)](https://liquibase.jira.com/wiki/spaces/LB/pages/1274937609/How+to+Write+Liquibase+Core+Unit+Tests)
* \[ ] Added [Integration Test(s)](https://liquibase.jira.com/wiki/spaces/LB/pages/1276608569/How+to+Write+Liquibase+Core+Integration+Tests)
* \[ ] Documentation Updated

#### Links
* Fixed Issue: [https://liquibase.jira.com/browse/CORE-3028](https://liquibase.jira.com/browse/CORE-3028)
* Local Branch: [https://github.com/liquibase/liquibase/tree/Plunts-CORE-3028](https://github.com/liquibase/liquibase/tree/Plunts-CORE-3028)
* Unit Tests: [https://github.com/liquibase/liquibase/pull/1590/checks](https://github.com/liquibase/liquibase/pull/1590/checks)
* Integration Tests: (shown in Unit Tests link above)
* Functional Tests: [https://jenkins.datical.net/job/liquibase-pro/job/Plunts-CORE-3028/](https://jenkins.datical.net/job/liquibase-pro/job/Plunts-CORE-3028/)

#### Testing
* Steps to Reproduce: [https://liquibase.jira.com/browse/CORE-3028](https://liquibase.jira.com/browse/CORE-3028)
* Guidance: 
  * UpdateSQL will not show a difference. Need to actually run update
  * Fix is independent of the database type. I tested using postgresql. Oracle, MSSQL, H2 were also reported as broken

#### Dev Verification
Replicated error pre-fix, verified error no longer occurs post-fix.

```java
Starting Liquibase at 16:07:16 (version 4.4.1-local-SNAPSHOT #0 built at 2021-06-30 20:47+0000)
Liquibase Version: 4.4.1-local-SNAPSHOT
Liquibase Pro 4.4.1-local-SNAPSHOT by Datical licensed to Liquibase Pro Customer until Tue Oct 26 19:00:00 CDT 2021
Skipping auto-registration
Liquibase command 'update' was executed successfully.
```

is shown in either case, but after the fix, `select * from table_with_clob` returns a not-null modified_time column.

## Test Requirements (Internal Liquibase QA)
* No new functional automated tests required (unit tests added are sufficient)

**Manual Test Requirements**

Use the attached changelog and text file to run tests. There are four changesets; two have dbms=mssql, and two have dbms=oracle. Execute test cases against **both** SQL Server and Oracle instances.

_Verify update is successful to create a table with clob and timestamp columns._

* COMMAND: liquibase update --changelog-file lb-1052-changelog.xml --labels lb1052
  * MSSQL
  * Oracle

_Verify insert of computed timestamp is not null for column MODIFIED_TIME._

* MSSQL
  * SQL SERVER: SELECT * FROM TABLE_WITHCLOB;
* ORACLE
  * ORACLE: SELECT * FROM TABLE_WITH_CLOB2;

┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-1052) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by molivasdat** [comment]
Hi @Plunts 
Thanks for your pull request!

Here’s what happens next:

A member of the Liquibase team will take a look at your contribution and may suggest: 

- Changes based on [guidelines for PR submission](https://liquibase.jira.com/wiki/spaces/LB/pages/1274904896/How+to+Contribute+Code+to+Liquibase+Core)
- Additional tests ([unit ](https://liquibase.jira.com/wiki/spaces/LB/pages/1274937609/How+to+Write+Liquibase+Core+Unit+Tests)and/or [integration](https://liquibase.jira.com/wiki/spaces/LB/pages/1276608569/How+to+Write+Liquibase+Core+Integration+Tests)) 
- Provide other feedback, like requesting changes. 

The PR will be prioritized according to our internal development and testing capacity.

We’ll let you know when it’s ready to move to the next step or if any changes are needed.

File: liquibase-core/src/test/java/liquibase/statement/InsertExecutablePreparedStatementTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        // when
+        statement.execute(preparedStatementFactory);
+
+        // then
+        verify(connection).prepareStatement("INSERT INTO DATABASECHANGELOG(MD5SUM, DATEEXECUTED) VALUES(?, GETDATE())");
+		verify(ps).setString(1, "7:e27bf9c0c2313160ef960a15d44ced47");
+		verify(ps, never()).setNull(eq(2), anyInt());
+	}
+
+//    @Test
```
Comment (nvoxland) [inline liquibase-core/src/test/java/liquibase/statement/InsertExecutablePreparedStatementTest.java:74]: Why did you end up commenting out these new tests?

**Review by nvoxland — CHANGES_REQUESTED** [review]

File: liquibase-core/src/test/java/liquibase/statement/InsertExecutablePreparedStatementTest.java (hunk 1 above)
Comment (Plunts) [reply liquibase-core/src/test/java/liquibase/statement/InsertExecutablePreparedStatementTest.java:74]: I just copied UpdateExecutablePreparedStatementTest and changed the first test to demonstrate the bug I wanted to fix. The second test is also commented there, I didn't even think about that, sorry.

**Review by nvoxland — APPROVED** [review]
Changes look good, thanks

**Review by suryaaki2 — APPROVED** [review]

**Pull request merged**